yew/src/virtual_dom/vcomp.rs

```diff
@@ -92,6 +97,19 @@ where
     }
 }
 
+#[cfg(feature = "ssr")]
+impl ToHtmlString for VComp {
+    fn to_html_string(&self) -> String {
+        match &self.scope {
+            None => "".to_string(),
```

Comment:
Should these be empty strings or should we return an `Err`?

Reply:
I would suggest that instead of a separate trait `ToHtmlString` we instead implement `TryInto<String>` for all the types. There's some [documentation from the standard libary](https://doc.rust-lang.org/stable/std/convert/trait.TryInto.html) about this which might useful.
I would suggest that an error type (using `thiserror` is also declared) and set as the associated error type.
Happy to provide help/asisstance if necessary.

Reply:
Is HTML the canonical way you would want to represent a VNode though? I would think the canonical way would just be the way it appears in source.

Reply:
I'm not 100% sure what you mean by "canonical" – would you mind elaborating.

Reply:
The word is ill-defined, but generally, once we decide that this is `the` way to convert VNodes to strings, it's hard to backtrack. I would think a fringe feature like SSR should not monopolize the standard Rust traits for core Yew structures; for example, what if Yew later goes the direction of something like React-Native where there really isn't HTML to speak of, as an output? Maybe output would instead be in the form of manipulating a native widget. Semantically, turning VNodes into strings doesn't seem like it should need anything to do with HTML for this reason.

Reply:
I haven't heard of anyone trying to prerender mobile applications. I suspect this is because a mobile app is already installed and downloaded when the user opens it, whereas a webpage has to load lots of code from the server and then start diffing – that's the problem which SSR (IMO) solves – it speeds up the loading of HTML pages.
This isn't a very major change; given that the traits are trying to do similar things – i.e. convert from virtual DOM types into a `String` of HTML content. 
Given that React Native bridges to native components on a device, it's unlikely that it would even be *possible* to prerender components on the server side.
React Native does have a way of building projects for web applications as well as mobile applications, but the same approach should work for rendering "native" components compiled for the web on the server as for rendering components designed for the web on the server.
Having a new trait called `ToHtmlString` suffers from the same disadvantages of `Into`/`From` or `TryInto`/`TryFrom`.
I hope this helps – let me know if anything here needs clarification.

Reply:
I agree you would never want to (pre-)render HTML on a mobile application -- that was actually my point, is that I think the Yew core (which is mostly useful for its paradigms around managing of state, communication between components, tree construction, etc.) is of a different concern from any particular runtime (e.g. HTML or in the future, perhaps Qt or native widgets). I would certainly like to use Yew in the future to build declarative non-web UI apps using Rust, at least...

Still not sure where (or whether) we are in disagreement on the above, but zooming out a little bit, what would be the downside of having a separate trait? Or alternatively, an upside to using `TryInto`? My main experience with `.into()` is that it can make the source code hard to read because it can be unclear what type you are converting into without using an IDE or something like Rust Analyzer, and even then it doesn't always get the type correctly surfaced to the developer so I would be curious what the advantage is.

Reply:
Sorry, posted from wrong account before, re-posting:

I agree you would never want to (pre-)render HTML on a mobile application -- that was actually my point, is that I think the Yew core (which is mostly useful for its paradigms around managing of state, communication between components, tree construction, etc.) is of a different concern from any particular runtime (e.g. HTML or in the future, perhaps Qt or native widgets). I would certainly like to use Yew in the future to build declarative non-web UI apps using Rust, at least...

Still not sure where (or whether) we are in disagreement on the above, but zooming out a little bit, what would be the downside of having a separate trait? Or alternatively, an upside to using TryInto? My main experience with .into() is that it can make the source code hard to read because it can be unclear what type you are converting into without using an IDE or something like Rust Analyzer, and even then it doesn't always get the type correctly surfaced to the developer so I would be curious what the advantage is.

Reply:
The type parameter on `into()` normally can't be inferred by the compiler and has to be explicity given. If `TryFrom<Html>` was implemented for `String` you get `into` for free, can use `String::from(<html>)` which gets rid of `Into` related error messages which some people may find hard to read and it's a more natural fit onto Rust applications than adding an additional trait.

Reply:
Thanks for the discussion, I will attempt to implement this in the next revision

Reply:
Looking at this now, and I can't find how to do what you were saying. The compiler does not seem happy with
```
48 |             VNode::VTag(vtag) => vtag.try_into::<Html>()?,
   |                                                  ^^^^ unexpected type argument
```

If I omit Html, the compiler is fine, but I would prefer to be explicit.

Reply:
The annotation would generally go on the left hand side, e.g.
```
let string: String = vtag.try_into()?; 
```